### ISSUE TITLE
Moderate: Some suggestions

### ISSUE THREAD

**Issue by MaximilianAnzinger** [ro-i/tumcsbot#1]
As an instructor, I'm interested in using the new moderate feature.

First of all, I'm not quite sure if I'm just not able to find it or if it doesn't exist. I'd like to see some proper documentation.

Here are some suggestions to improve the functionality:

1. For tutors we've usually a private group. We have to add them manually when creating the stream and once again to configure the moderation groups.
- Add all users of a stream to a group
- OR Create stream with all users of group

2. Permissions for every stream have to be granted individually.
- Match by prefix (similar to auto subscription for streams, i.e. PGdP)
- MUST HAVE: Exclusions (e.g. announcement streams)

3. Tutors should not be able to delete messages from instructors. This can happen by accident.

4. Are the messages deleted from the DB? What if a message is deleted accidentally or what if a student complains that the message shouldn't have been deleted? Can we undo the delete?

5. I assume the settings will be (almost) the same every semester.
- Export settings to a file so we can import them again next year.

6. Should tutors have access to the feature once their employment ends?
- Configurable timeframe when the moderation group is "active"

**Comment by rotmanjanez** [comment]
Thank you for your feedback @MaximilianAnzinger.
Regarding Q2: It was a design choice to not have regex in a command to prevent accidental arguments. There is a plugin in progress, where one can list streams by regex. Then this list can be copy-pasted for any other command. Is this workflow  suitable for you?

Regarding Q4: Yes, the messages are deleted from the db (no recovery possible), however, it is possible the send a copy of the original message to the student, so it can be easily be re-posted (also useful if only a portion of the message should be changed). Is this sufficient as a "re-do" feature, or is there the need for a real re-do?

**Comment by MaximilianAnzinger** [comment]
> Regarding Q2: It was a design choice to not have regex in a command to prevent accidental arguments. There is a plugin in progress, where one can list streams by regex. Then this list can be copy-pasted for any other command. Is this workflow suitable for you?

Not really. We create new streams for exercises each week. This would require me to update the configuration every week.

> Regarding Q4: Yes, the messages are deleted from the db (no recovery possible), however, it is possible the send a copy of the original message to the student, so it can be easily be re-posted (also useful if only a portion of the message should be changed). Is this sufficient as a "re-do" feature, or is there the need for a real re-do?

Seems kind of okayish. However, that would disturb the flow of the conversation if the main question gets deleted and then added at the end/middle again.
